### PR TITLE
Source code linting

### PR DESCRIPTION
**Pull request by jaytmiller**
This PR adds flake8 Python source code scanning to our GitHub Action workflow.
Existing code was adjusted with minor changes like dropping unused imports and whitespace formatting.

### CODE REVIEW

File: deployments/common/image/common-scripts/set-notebook-kernel [hunk 1]
```diff
@@ -22,28 +22,35 @@ def set_kernel(kernel, display_name, notebook):
...
 def main():
     args = parse_args()
     with open(f"/opt/environments/{args.kernel}/kernel.name") as f:
         display_name = f.read().strip()
     for notebook in args.notebooks:
         if not notebook.startswith("#") and notebook.endswith(".ipynb"):
-            print(f"Setting kernel={args.kernel}  display_name={display_name}  notebook={notebook}")
+            print(
+                f"Setting kernel={args.kernel}  display_name={display_name}  notebook={notebook}"
+            )
```
Comment (cslocum) [inline deployments/common/image/common-scripts/set-notebook-kernel:46]: Is this type of nesting enforced by flake8?

File: deployments/common/image/common-scripts/test-notebooks [hunk 2]
```diff
@@ -29,7 +29,11 @@ def test_notebooks(kernel, notebooks, jobs=1):
 
     Return   count of failed notebooks
     """
-    print(divider(f"Testing {len(notebooks)} notebooks on kernel {kernel} using {jobs} jobs").strip())
+    print(
+        divider(
+            f"Testing {len(notebooks)} notebooks on kernel {kernel} using {jobs} jobs"
+        ).strip()
+    )
```
Comment (cslocum) [inline deployments/common/image/common-scripts/test-notebooks:36]: Same questioning here, with the nesting.

**Review by cslocum — APPROVED** [review]
Looks great!  Definitely much cleaner :+1:

**Review by jmatuskey — APPROVED** [review]

File: deployments/common/image/common-scripts/set-notebook-kernel (hunk 1 above)
Comment (jaytmiller) [reply deployments/common/image/common-scripts/set-notebook-kernel:46]: No.   That nesting was generated by the "black" source code formatter I ran on the script to take care of other whitespace issues flake8 did care about.

**Pull request merged**